### ISSUE TITLE
element.matches() can be array of strings

### ISSUE THREAD

**Issue by Criiis** [microsoft/TypeScript#49454]
# Bug Report

### 🔎 Search Terms

Element.matches()
Argument of type 'string[]' is not assignable to parameter of type 'string'.

### 🕗 Version & Regression Information

element.matches() missing array of strings as a type.
### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.7.2#code/CYewxgrgtgpgdgFwHQCMTAJ5KgQwWACxgGcAKAciQBUMAHGAZTACcBLWhcgSgChRJYiVOiy58RMgG1KNekzYdyAGgAElVsWVqA5sxh5yAXS5A)

### 💻 Code

```ts
document.body.matches('.TypeScript')
document.body.matches(['.TypeScript', '.is', 'great'])
```

### 🙁 Actual behavior
type atm is: (method) Element.matches(selectors: string): boolean
The argument of type 'string[]' is not assignable to the parameter of type 'string'.

### 🙂 Expected behavior
it shouldn't give any error if the parameter is an array of strings
type should be: (method) Element.matches(selectors: string | string[]): boolean

**Comment by Criiis** [comment]
https://developer.mozilla.org/en-US/docs/Web/API/Element/matches

**Comment by RyanCavanaugh** [comment]
I don't agree with this (or the edit you made to MDN for that matter). The spec has no mention of how to handle an array; this reason this appears to work is because the built-in `toString` behavior of `Array` does a comma-delimited join, which happens to be the same delimiter as CSS, so it works by coincidence.

You can try this in the console:
```js
> class MyArray extends Array { toString() { return "wrong" } }
> document.body.matches(new MyArray("body"))
false
```

**Comment by Criiis** [comment]
Hey @RyanCavanaugh, Thank you so much for the explanation.
That makes sense, and is also very interesting, cause if I do:
`document.body.matches(['.logged-in', '.env-production'])` which from what you are saying it will be the same as `document.body.matches('.logged-in,.env-production')` because of the `toString`  behavior, and yes, it gives me exactly the same result, amazing.
Thank you!
